templates/examples/extensions/py_lambda_hooks/CanvasHook/CanvasHook.py

```diff
@@ -0,0 +1,405 @@
+
+        course_name = []
+        if courses: 
+            # Loop through the courses.
+            for course in courses:
+                course_name.append(course.name)
+
+    result = {"CourseNames": course_name}
+    return result
+    
+
+#----------------------------------------------------------------------
+# function: query_choices_for_student
+#----------------------------------------------------------------------
+def query_choices_for_student(canvas, student_user_name, userinput):
+    # Get the user using user_id to match with LMS SIS_ID
+    user = getCanvasUser (canvas, student_user_name)
+    
+    if user:
+        courses = user.get_courses(enrollment_status='active')
+        if courses: 
+            course_name = []
+            # Loop through the courses.
+            for course in courses:
+                course_name.append(course.name)
+            choice = process.extractOne(userinput, course_name, scorer=fuzz.token_set_ratio)
+
+    result = {"Choice": choice[0]}
+    return result
+
+
+#----------------------------------------------------------------------
+# function: query_course_assignments_for_student
+#----------------------------------------------------------------------
+def query_course_assignments_for_student(canvas, student_user_name, userinput):
+    course_assignments = ''
+    blnHasAssignments = False
+
+    # Get the user using user_id to match with LMS SIS_ID
+    user = getCanvasUser (canvas, student_user_name)
+    
+    if user:
+        course_list = []
+        courses = user.get_courses(enrollment_status='active')
+    
+        # Loop through the courses.
+        for course in courses:
+            course_list.append(course.name)
+        choice = process.extractOne(userinput, course_list, scorer=fuzz.token_set_ratio)
+
+        for course in courses:
+            blnHasAssignments = False
+            #check for matching userinput with course names
+            if course.name == choice[0] and choice[1] > MATCHING_TOLERANCE_SCORE:
+                course_assignments = "<b>" + course.name + "</b> <ul>"
+                # Loop through the assignments that have not been submitted
+                for assignment in course.get_assignments(bucket='unsubmitted'):
+                    blnHasAssignments = True
+                    if assignment.due_at:
+                        due_date = datetime.datetime.strptime(assignment.due_at,'%Y-%m-%dT%H:%M:%SZ')
+                        due_date_string = '{0}, {1} {2}, {3}'.format(calendar.day_name[due_date.weekday()], due_date.strftime("%B"), due_date.strftime("%-d"), due_date.strftime("%Y"))
+                        course_assignments += "<li>{} -- is due: {}</li>".format(assignment.name, due_date_string)
+                    else:
+                        course_assignments += "<li>{} -- has no due date</li>".format(assignment.name)
+                break
+            else:
+                # Loop through the assignments that have not been submitted
```

Comment:
Similar to if case, can it be simplified by refactoring and removing duplication?